e2e/e2e_test.go

```diff
@@ -0,0 +1,172 @@
+		SetAccountID(receiving).
+		Execute(client)
+
+	fmt.Println(fmt.Sprintf(`HBAR custodian balance after transaction: [%d]`, recieverBalanceNew.Hbars.AsTinybar()))
+
+	// Verify that the custodial address has recieved exactly the amount sent
+	if (recieverBalanceNew.Hbars.AsTinybar() - recieverBalance.Hbars.AsTinybar()) != hedera.HbarFrom(hBarAmount, "hbar").AsTinybar() {
+		t.Fatal(`Expected to recieve the exact transfer amount of hbar`)
+	}
+
+	// Subscribe to Topic
+	hedera.NewTopicMessageQuery().
+		SetStartTime(time.Unix(0, time.Now().UnixNano())).
+		SetTopicID(topicID).
+		Subscribe(
+			client,
+			func(response hedera.TopicMessage) {
+				msg := &validatorproto.TopicSubmissionMessage{}
+				proto.Unmarshal(response.Contents, msg)
+
+				if msg.GetType() == validatorproto.TopicSubmissionType_EthSignature {
+					//Verify that all the submitted messages have signed the same transaction
+					if msg.GetTopicSignatureMessage().TransactionId != fromHederaTransactionID(transactionResponse.TransactionID) {
+						t.Fatal(`Expected signature message to contain the transaction id`)
+					}
+					ethSignaturesCollected++
+				}
+
+				if msg.GetType() == validatorproto.TopicSubmissionType_EthTransaction {
+					//Verify that the eth transaction message has been submitted
+					if msg.GetTopicEthTransactionMessage().TransactionId != fromHederaTransactionID(transactionResponse.TransactionID) {
+						t.Fatal(`Expected ethereum transaction message to contain the transaction id`)
+					}
+					ethTransactionHash = msg.GetTopicEthTransactionMessage().GetEthTxHash()
+					ethTransMsgCollected++
+				}
+			},
+		)
+
+	// Wait for topic consensus messages to arrive
+	time.Sleep(60 * time.Second)
+
+	// Check that all the validators have submitted a message with authorisation signature
+	if ethSignaturesCollected != validatorsCount {
+		t.Fatal(`Expected the count of collected signatures to equal the number of validators`)
+	}
+
+	// Verify the exactly on eth transaction hash has been submitted
+	if ethTransMsgCollected != 1 {
+		t.Fatal(`Expected to submit exaclty 1 ethereum transaction in topic`)
+	}
+
+	success, err := ethClient.WaitForTransactionSuccess(common.HexToHash(ethTransactionHash))
+	// Verify that the eth transaction has been mined and succeeded
+	if success == false {
+		t.Fatal(`Expected to mine successfully the broadcasted ethereum transaction`)
+	}
+
+	// Get the wrapped hbar balance of the receiver after the transfer
+	whbarBalanceAfter, err := whbarInstance.BalanceOf(&bind.CallOpts{}, whbarReceiverAddress)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("WHBAR balance after transaction: [%s]\n", whbarBalanceAfter)
+
+	// Verify that the ethereum address hass recieved the exact transfer amount of WHBARs
+	if (whbarBalanceAfter.Int64() - whbarBalanceBefore.Int64()) != hedera.HbarFrom(hBarAmount, "hbar").AsTinybar() {
+		t.Fatal(`Expected to recieve the exact transfer amount of WHBAR`)
+	}
+
```

Comment:
Thoughts on db records assertion in the end?
@Daniel-K-Ivanov

Reply:
That might be a nice addition on top of the current verifications, but I guess we could include it in a separate PR.